tests/models/test_anthropic.py

```diff
@@ -284,6 +285,148 @@ async def test_async_request_prompt_caching(allow_model_requests: None):
     assert last_message.cost().total_price == snapshot(Decimal('0.00002688'))
 
 
+async def test_cache_point_adds_cache_control(allow_model_requests: None):
+    """Test that CachePoint correctly adds cache_control to content blocks."""
+    c = completion_message(
+        [BetaTextBlock(text='response', type='text')],
+        usage=BetaUsage(input_tokens=3, output_tokens=5),
+    )
+    mock_client = MockAnthropic.create_mock(c)
+    m = AnthropicModel('claude-haiku-4-5', provider=AnthropicProvider(anthropic_client=mock_client))
+    agent = Agent(m)
+
+    # Test with CachePoint after text content
+    await agent.run(['Some context to cache', CachePoint(), 'Now the question'])
+
+    # Verify cache_control was added to the right content block
+    completion_kwargs = get_mock_chat_completion_kwargs(mock_client)[0]
+    messages = completion_kwargs['messages']
+    assert len(messages) == 1
```

Comment:
Please use `snapshot()` as much as possible! I want to see the entire message structure and not have to parse all these assertions